--- lib/functions	2020-02-08 10:48:07.000000000 +0100
+++ lib/functions.lsmod	2020-03-13 10:59:06.029459498 +0100
@@ -1106,6 +1106,9 @@

 # module is needed (ie. is requested, is available and isn't loaded already)
 is_module() {
+	# return false if /sbin/lsmod not exists (like in containers)
+	[ -x /sbin/lsmod ] || return 1
+
 	# module name without .o at end
 	if ! lsmod | grep -q "$1"; then
 		if ls -1R /lib/modules/$(uname -r)/ 2> /dev/null | grep -q "^${1}.\(\|k\)o\(\|.gz\)"; then
