diff -ru rc-scripts-0.3.1/rc.d/init.d/functions.network rc-scripts-0.3.1-AX/rc.d/init.d/functions.network
--- rc-scripts-0.3.1/rc.d/init.d/functions.network	Sat Nov  9 00:41:49 2002
+++ rc-scripts-0.3.1-AX/rc.d/init.d/functions.network	Mon Nov 19 20:07:17 2001
@@ -328,7 +328,7 @@
 	if [ -z "${GATEWAYDEV}" -o "${GATEWAYDEV}" = "${DEVICE}" ]; then
 	    # set up default gateway
 	    if [ -n "${GATEWAY}" ]; then
-	  	ip -4 route add default via ${GATEWAY} dev ${DEVICE} ${IP4SRCADDR} ${TNLOPTS}
+	  	ip -4 route add default via ${GATEWAY} dev ${DEVICE} ${IP4SRCADDR} ${TNLOPTS} onlink
 	    	DEFGW=${GATEWAY}
 	    elif [ "${GATEWAYDEV}" = "${DEVICE}" ]; then
 	 	ip -4 route add default dev ${DEVICE} ${IP4SRCADDR} ${TNLOPTS}
@@ -342,7 +342,7 @@
     	    if [ -z "${GATEWAY6DEV}" -o "${GATEWAY6DEV}" = "${DEVICE}" ]; then
 	    	# set up default gateway
 	    	if [ -n "${GATEWAY6}" ]; then
-	    		ip -6 route add default via ${GATEWAY6} dev ${DEVICE} ${IP6SRCADDR}
+	    		ip -6 route add default via ${GATEWAY6} dev ${DEVICE} ${IP6SRCADDR} onlink
 			DEFGW6=${GATEWAY6}
 	    	elif [ "${GATEWAY6DEV}" = "${DEVICE}" ]; then
 	    		ip -6 route add default dev ${DEVICE} ${IP6SRCADDR}
Only in rc-scripts-0.3.1/rc.d/init.d: functions.network~
