--- rc-scripts-0.4.0.20/rc.d/rc.sysinit.orig	2005-08-23 18:22:13.000000000 +0200
+++ rc-scripts-0.4.0.20/rc.d/rc.sysinit	2005-09-19 01:12:32.000000000 +0200
@@ -69,6 +69,11 @@
 	[ -n "$LC_ALL" ] && export LC_ALL || unset LC_ALL
 	[ -n "$LANGUAGE" ] && export LANGUAGE || unset LANGUAGE
 	[ -n "$LINGUAS" ] && export LINGUAS || unset LINGUAS
+	case "$LANG" in
+		*.utf8|*.UTF-8)
+			[ -x /bin/unicode_start ] && /sbin/consoletype fg && /bin/unicode_start
+		;;
+	esac
 fi
 
 # Read network config data
