Index: sysconfig/network-scripts/ifup-aliases
===================================================================
--- sysconfig/network-scripts/ifup-aliases	(revision 10197)
+++ sysconfig/network-scripts/ifup-aliases	(working copy)
@@ -55,7 +55,9 @@
 	if echo $IP_ADDR | grep -q ":"; then
 		if is_yes "${IPV6_NETWORKING}"; then
 			ip -6 addr add ${IP_ADDR} dev ${DEVICE} ${IP_ADDR_OPT}
-			ip -6 addrlabel add prefix ${IP_ADDR} dev ${DEVICE} label ${IP_ADDR_LABEL}
+			if [ -n "${IP_ADDR_LABEL}" ]; then
+				ip -6 addrlabel add prefix ${IP_ADDR} dev ${DEVICE} label ${IP_ADDR_LABEL}
+			fi
 		fi
 	elif is_yes "${IPV4_NETWORKING}"; then
 		# If broadcast is missing then autocalculate it
