Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(wersja 12628)
+++ rc.d/rc.sysinit	(kopia robocza)
@@ -335,7 +335,7 @@
 		fi
 	fi
 
-	if [ "$usb" = "1" ] && ! is_fsmounted usbfs /proc/bus/usb ; then
+	if [ "$usb" = "1" -a -d /proc/bus/usb ] && ! is_fsmounted usbfs /proc/bus/usb ; then
 		run_cmd "Mounting USB filesystem" mount -n -t usbfs -o devgid=78,devmode=664 usbfs /proc/bus/usb
 	fi
 
