diff -urN rc-scripts-0.3.1.orig/README rc-scripts-0.3.1/README
--- rc-scripts-0.3.1.orig/README	Tue May 15 18:05:58 2001
+++ rc-scripts-0.3.1/README	Wed Nov 27 16:31:22 2002
@@ -36,5 +36,5 @@
 use our Bug Tracking System ( http://bugs.pld.org.pl/ ).
 
 -- 
-PLD GNU/Linux rc-scripts Team,
+PLD Linux rc-scripts Team,
 
diff -urN rc-scripts-0.3.1.orig/po/pl.po rc-scripts-0.3.1/po/pl.po
--- rc-scripts-0.3.1.orig/po/pl.po	Wed Nov 27 16:28:22 2002
+++ rc-scripts-0.3.1/po/pl.po	Wed Nov 27 16:31:07 2002
@@ -553,8 +553,8 @@
 msgstr "Ustawianie statycznych wpisów RARP"
 
 #: ./rc.d/rc.sysinit:55
-msgid "\t\t\t%sPowered by %sPLD GNU/Linux%s"
-msgstr "\t\t\t%sNapedzane przez %sPLD GNU/Linux%s"
+msgid "\t\t\t%sPowered by %sPLD Linux%s"
+msgstr "\t\t\t%sNapedzane przez %sPLD Linux%s"
 
 #: ./rc.d/init.d/functions:110
 msgid "ERROR: Networking is down. %s can't be run."
diff -urN rc-scripts-0.3.1.orig/po/rc-scripts.pot rc-scripts-0.3.1/po/rc-scripts.pot
--- rc-scripts-0.3.1.orig/po/rc-scripts.pot	Sun Jan 27 05:04:18 2002
+++ rc-scripts-0.3.1/po/rc-scripts.pot	Wed Nov 27 16:30:53 2002
@@ -576,7 +576,7 @@
 msgstr ""
 
 #: ../rc.d/rc.sysinit:55
-msgid "\t\t\t%sPowered by %sPLD GNU/Linux%s"
+msgid "\t\t\t%sPowered by %sPLD Linux%s"
 msgstr ""
 
 #: ../rc.d/init.d/functions:123
diff -urN rc-scripts-0.3.1.orig/rc.d/init.d/functions.network rc-scripts-0.3.1/rc.d/init.d/functions.network
--- rc-scripts-0.3.1.orig/rc.d/init.d/functions.network	Mon Nov 19 20:07:17 2001
+++ rc-scripts-0.3.1/rc.d/init.d/functions.network	Wed Nov 27 16:31:12 2002
@@ -140,7 +140,7 @@
 fi
 }
 
-# PLD GNU/Linux network functions
+# PLD Linux network functions
 # Copyright (C) 1999, 2000 Arkadiusz Mi¶kiewicz <misiek@pld.org.pl>
 
 # Setup Network Address Translation (NAT)
diff -urN rc-scripts-0.3.1.orig/rc.d/rc.sysinit rc-scripts-0.3.1/rc.d/rc.sysinit
--- rc-scripts-0.3.1.orig/rc.d/rc.sysinit	Sun Jan 27 03:13:50 2002
+++ rc-scripts-0.3.1/rc.d/rc.sysinit	Wed Nov 27 16:31:18 2002
@@ -53,7 +53,7 @@
 fi
 
 # Print welcome message
-nls "\t\t\t%sPowered by %sPLD GNU/Linux%s" "$(termput setaf 6)" "$(termput setaf 2)" "$(termput setaf 7)"
+nls "\t\t\t%sPowered by %sPLD Linux%s" "$(termput setaf 6)" "$(termput setaf 2)" "$(termput setaf 7)"
 if is_yes "$PROMPT"; then
 	nls -n "\t\t  Press %sI%s to enter interactive startup." "$(termput setaf 1)" "$(termput setaf 7)"
 	echo
