--- rc-scripts-0.4.2.4/modules~	2009-08-16 22:49:36.000000000 +0200
+++ rc-scripts-0.4.2.4/modules	2009-11-08 09:37:35.299866086 +0100
@@ -8,6 +8,6 @@
 # Comments begin with a `#', and everything on the line after them are ignored.
 #
 # You can also have /etc/modules.X.Y and /etc/modules.X.Y.Z, where X.Y.Z are kernel version numbers
-# The modules files are preocessed in exactly that order too.
+# The modules files are processed in exactly that order too.
 
 #loop
