diff -ur rc-scripts-0.4.0.25/rc.d/init.d/functions rc-scripts/rc.d/init.d/functions
--- rc-scripts-0.4.0.25/rc.d/init.d/functions	2005-11-26 16:15:22.000000000 +0100
+++ rc-scripts/rc.d/init.d/functions	2006-01-20 16:40:17.000000000 +0100
@@ -392,8 +392,13 @@
 # - inside chroot get only those processes, which are inside chroot.
 # - don't filter out pids which do not have corresponding running processes (process died etc)
 # (note: some processes like named are chrooted but run outside chroot)
+# - do nothing inside vserver
 filter_chroot()
 {
+	if is_yes "$VSERVER"; then
+		echo $@
+		return
+	fi
 	if [ $# -lt 1 -o ! -d /proc/1 ] ; then
 		echo $@
 		return
