diff -durN rc-scripts-0.3.1.orig/rc.d/rc.shutdown rc-scripts-0.3.1/rc.d/rc.shutdown
--- rc-scripts-0.3.1.orig/rc.d/rc.shutdown	Sun Jan 27 03:13:29 2002
+++ rc-scripts-0.3.1/rc.d/rc.shutdown	Fri Nov 14 10:08:21 2003
@@ -90,6 +90,11 @@
 if [ "$runlevel" = "0" ] ; then
        	show "The system is halted"; ok
 	[ -f /fastboot ] && (show "On the next boot fsck will be skipped."; ok)
+	
+	if [ -x /sbin/poweroff-ups -a -f /etc/killpower -a -f /etc/sysconfig/ups ] ; then
+	    . /etc/sysconfig/ups
+	    is_yes "$POWEROFF_UPS" && /sbin/poweroff-ups
+	fi
 	eval halt -d -p
 else
 	show "Please stand by while rebooting the system"; ok
diff -durN rc-scripts-0.3.1.orig/rc.d/rc.sysinit rc-scripts-0.3.1/rc.d/rc.sysinit
--- rc-scripts-0.3.1.orig/rc.d/rc.sysinit	Fri Nov 14 09:27:20 2003
+++ rc-scripts-0.3.1/rc.d/rc.sysinit	Fri Nov 14 10:08:20 2003
@@ -574,7 +574,7 @@
 fi
 
 # Clean up /etc.
-rm -f /fastboot /fsckoptions /forcefsck /halt /poweroff
+rm -f /fastboot /fsckoptions /forcefsck /halt /poweroff /etc/killpower
 
 # Clean up /var
 # I'd use find, but /usr may not be mounted.
