--- rc-scripts-0.4.15/rc.d/rc.sysinit.orig	2016-06-19 20:33:41.190705364 +0200
+++ rc-scripts-0.4.15/rc.d/rc.sysinit	2016-06-19 20:51:05.967360163 +0200
@@ -124,10 +124,11 @@
 	if ! grep -Fq selinuxfs /proc/filesystems; then
 		# no support in kernel, no chance
 		SELINUX=no
+		return
 	fi
 
-	if ! is_fsmounted selinuxfs /selinux; then
-		mount -n -o gid=17 -t selinuxfs selinuxfs /selinux
+	if ! is_fsmounted selinuxfs $selinuxfsdir; then
+		mount -n -o gid=17 -t selinuxfs selinuxfs $selinuxfsdir
 	fi
 
 	# Check SELinux status
@@ -424,6 +425,12 @@
 		mount -n -o gid=17 -t securityfs securityfs /sys/kernel/security
 	fi
 
+	if [ -d /sys/fs/selinux ]; then
+		# Linux 3.0+
+		selinuxfsdir=/sys/fs/selinux
+	else
+		selinuxfsdir=/selinux
+	fi
 	init_selinux
 
 	# PLD Linux LiveCD support
@@ -695,8 +702,8 @@
 		fi
 	fi
 
-	if is_fsmounted selinuxfs /selinux; then
-		mount -f -t selinuxfs selinuxfs /selinux 2> /dev/null
+	if is_fsmounted selinuxfs $selinuxfsdir; then
+		mount -f -t selinuxfs selinuxfs $selinuxfsdir 2> /dev/null
 	fi
 
 	if [ ! -f /proc/modules ]; then
