--- etc/rc.d/init.d/network~	2005-10-16 04:22:43.823119234 +0300
+++ etc/rc.d/init.d/network	2005-10-16 04:22:45.702829164 +0300
@@ -57,7 +57,9 @@
 modprobe_net
 
 # Setup interfaces names
-[ -x /sbin/nameif -a -f /etc/mactab ] && run_cmd "Setting interfaces names (nameif)" /sbin/nameif
+if [ -x /sbin/nameif -a -f /etc/mactab ] && [ $(egrep -v '^(#| *$)' /etc/mactab | wc -l) -gt 0 ]; then
+    run_cmd "Setting interfaces names (nameif)" /sbin/nameif
+fi
 
 # Kernel network parameters
 sysctl -e -p /etc/sysctl.conf > /dev/null 2>&1
