diff -ruN rc-scripts-0.4.0.27./rc.d/rc.sysinit rc-scripts-0.4.0.27/rc.d/rc.sysinit
--- rc-scripts-0.4.0.27./rc.d/rc.sysinit	2006-07-17 18:49:54.000000000 +0200
+++ rc-scripts-0.4.0.27/rc.d/rc.sysinit	2006-07-17 18:55:10.610862496 +0200
@@ -701,6 +701,11 @@
 		rc_splash "fsck stop"
 	fi
 
+	# FUSE module is not automatically loaded
+	if grep -q "fuse" /etc/fstab 2>/dev/null; then
+		run_cmd "Loading FUSE module" modprobe -s fuse
+	fi
+
 	# Mount all other filesystems (except for NFS and /proc, which is already
 	# mounted). Contrary to standard usage,
 	# filesystems are NOT unmounted in single user mode.
