Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(revision 6539)
+++ rc.d/rc.sysinit	(working copy)
@@ -377,14 +377,14 @@
 	done
 fi
 
-# Clear mtab
-:>/etc/mtab
-[ -f /etc/cryptomtab ] && :>/etc/cryptomtab
-
 # Remove stale backups
 rm -f /etc/mtab~ /etc/mtab~~ /etc/cryptomtab~ /etc/cryptomtab~~
 
 if ! is_yes "$VSERVER"; then
+	# Clear mtab
+	:>/etc/mtab
+	[ -f /etc/cryptomtab ] && :>/etc/cryptomtab
+
 	# Enter root, /proc, /sys, devfs and other into mtab.
 	mount -f /
 	mount -f /proc
Index: rc.d/init.d/functions
===================================================================
--- rc.d/init.d/functions	(revision 6539)
+++ rc.d/init.d/functions	(working copy)
@@ -52,7 +52,7 @@
 [ -z "$COLUMNS" ] && COLUMNS=80
 
 if [ -z "$VSERVER" -o "$VSERVER" = "detect" ]; then
-	_ctx="$(awk '/s_context:.*/ { print $2 }' /proc/self/status 2>/dev/null)"
+	_ctx="$(awk '/(s_context|VxID):.*/ { print $2 }' /proc/self/status 2>/dev/null)"
 	if [ -z "$_ctx" -o "$_ctx" = "0" ]; then
 		VSERVER=no
 	else
