--- ./sysconfig/interfaces/data/chat-ppp0.gprs-plusgsm.org	2005-06-27 11:27:55.299170088 +0200
+++ ./sysconfig/interfaces/data/chat-ppp0.gprs-plusgsm	2005-06-27 11:28:58.937495592 +0200
@@ -1,6 +1,9 @@
-ABORT BUSY
-ABORT ERROR
+ABORT "BUSY"
+ABORT "ERROR"
 ABORT "NO CARRIER"
+ABORT "NO DIALTONE"
+ABORT "NO ANSWER"
+ABORT "Username/Password Incorrect"
 "" AT
-OK-AT-OK ATD"$PHONE"
+OK-AT-OK ATDT"$PHONE"
 CONNECT
--- ./sysconfig/interfaces/ifcfg-ppp0.gprs-plusgsm.org	2005-06-27 11:27:06.924524152 +0200
+++ ./sysconfig/interfaces/ifcfg-ppp0.gprs-plusgsm	2005-06-27 11:27:31.687759568 +0200
@@ -16,8 +16,8 @@
 #ACTIVEFILTER=
 #IDLE=
 PHONE='*99***1#'
-USER=ppp
-PASSWORD=ppp
+USER=plusgsm
+PASSWORD=plusgsm
 #INITSTRING=ATe0s2=255
 DATAFORCHAT="USER PASSWORD INITSTRING PHONE"
 
