--- rc-scripts-0.4.5.2/rc.d/init.d/network.orig	2011-12-14 12:23:30.000000000 +0100
+++ rc-scripts-0.4.5.2/rc.d/init.d/network	2012-01-25 10:37:12.670299072 +0100
@@ -135,7 +135,8 @@
 				case $i in
 					*ifcfg-lo) continue ;;
 				esac
-				ONBOOT=""; . "$i" 2>/dev/null
+				ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+				[ ${USERS:-no} != no ] && continue
 				[ ${ONBOOT:-no} = yes ] && echo "${i##*/ifcfg-}"
 			done
 		`
@@ -145,7 +146,8 @@
 				case ${i##*/} in
 					ifcfg-lo|ifcfg-sit*|ifcfg-atm*|ifcfg-lec*|ifcfg-nas*|ifcfg-br*|ifcfg-*.*) continue ;;
 				esac
-				ONBOOT=""; . "$i" 2>/dev/null
+				ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+				[ ${USERS:-no} != no ] && continue
 				[ ${ONBOOT:-no} = yes ] && echo "${i##*/ifcfg-}"
 			done
 		`
@@ -156,7 +158,8 @@
 					ifcfg-*.*) ;;
 					*) continue ;;
 				esac
-				ONBOOT=""; . "$i" 2>/dev/null
+				ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+				[ ${USERS:-no} != no ] && continue
 				[ ${ONBOOT:-no} = yes ] && echo "${i##*/ifcfg-}"
 			done
 		`
@@ -167,7 +170,8 @@
 					ifcfg-br*) ;;
 					*) continue ;;
 				esac
-				ONBOOT=""; . "$i" 2>/dev/null
+				ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+				[ ${USERS:-no} != no ] && continue
 				[ ${ONBOOT:-no} = yes ] && echo "${i##*/ifcfg-}"
 			done
 		`
@@ -178,7 +182,8 @@
 					ifcfg-sit*) ;;
 					*) continue ;;
 				esac
-				ONBOOT=""; . "$i" 2>/dev/null
+				ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+				[ ${USERS:-no} != no ] && continue
 				[ ${ONBOOT:-no} = yes ] && echo "${i##*/ifcfg-}"
 			done
 		`
@@ -186,7 +191,8 @@
 
 	tunnels=$(
 		for i in $(network_interface_configs 'tnlcfg-*'); do
-			ONBOOT=""; . "$i" 2>/dev/null
+			ONBOOT=""; USERS=""; . "$i" 2>/dev/null
+			[ ${USERS:-no} != no ] && continue
 			[ ${ONBOOT:-no} = yes ] && echo "${i##*/tnlcfg-}"
 		done
 	)
