--- ./sysconfig/network-scripts/ifup.org	2005-12-21 13:05:10.392669624 +0100
+++ ./sysconfig/network-scripts/ifup	2005-12-21 13:08:51.244095104 +0100
@@ -147,7 +147,7 @@
 				;;
 			  dhcpcd)
 				[ -n "$DHCP_HOSTNAME" ] && DHCP_ARGS="-h $DHCP_HOSTNAME"
-				DHCP_ARGS="$DHCP_ARGS $DEVICE"
+				DHCP_ARGS="-d $DHCP_ARGS $DEVICE"
 				;;
 			  dhcpxd)
 				[ -n "$DHCP_HOSTNAME" ] && DHCP_ARGS="-H $DHCP_HOSTNAME"
