diff -durN -x '*~' rc-scripts-0.4.0.5.orig/src/Makefile.am rc-scripts-0.4.0.5/src/Makefile.am
--- rc-scripts-0.4.0.5.orig/src/Makefile.am	2003-12-07 16:58:12.000000000 +0100
+++ rc-scripts-0.4.0.5/src/Makefile.am	2004-12-19 14:51:54.862899648 +0100
@@ -2,6 +2,8 @@
 
 DEFS = @DEFS@ -D_GNU_SOURCE=1
 
+glibdir = /usr/lib
+
 sysconf_DATA = \
 	initlog.conf
 
@@ -53,7 +55,7 @@
 getkey_LDADD = -lpopt
 
 ppp_watch_SOURCES = ppp-watch.c shvar.c shvar.h
-ppp_watch_LDADD = /usr/lib/libglib.a
+ppp_watch_LDADD = $(glibdir)/libglib.a
 INCLUDES = `glib-config --cflags`
 
 consoletype_SOURCES = consoletype.c
