--- rc-scripts-0.3.1/rc.d/init.d/functions.network~	Wed Oct 30 16:39:15 2002
+++ rc-scripts-0.3.1/rc.d/init.d/functions.network	Wed Oct 30 16:39:15 2002
@@ -354,7 +354,7 @@
 	    	if [ -n "${IPV6_GLOBALROUTEGW}" ]; then
 	    	    ip -6 route add 2000::/3 via ${IPV6_GLOBALROUTEGW} dev ${DEVICE} ${IP6SRCADDR}
 	    	elif [ "${IPV6_GLOBALROUTEDEV}" = "${DEVICE}" ]; then
-	            ip -6 route add 2000::/3 via dev ${DEVICE} ${IP6SRCADDR}
+	            ip -6 route add 2000::/3 dev ${DEVICE} ${IP6SRCADDR}
 		fi
 	    fi
         fi
