--- rc-scripts-0.4.0.24/rc.d/init.d/sys-chroots~	Sun Jan 15 08:44:33 2006
+++ rc-scripts-0.4.0.24/rc.d/init.d/sys-chroots	Sun Jan 15 08:44:33 2006
@@ -76,12 +76,12 @@
 			if ! is_no "$NEED_XFILES" ; then
 				:>/var/run/utmpx
 				touch /var/log/wtmpx
-				chown root.utmp /var/run/utmpx /var/log/wtmpx
+				chown root:utmp /var/run/utmpx /var/log/wtmpx
 				chmod 0664 /var/run/utmpx /var/log/wtmpx
 			else
 				:>/var/run/utmp
 				touch /var/log/wtmp
-				chown root.utmp /var/run/utmp /var/log/wtmp
+				chown root:utmp /var/run/utmp /var/log/wtmp
 				chmod 0664 /var/run/utmp /var/log/wtmp
 			fi
 			# Clean /tmp
--- rc-scripts-0.4.0.24/rc.d/rc.sysinit~	Sun Jan 15 08:43:26 2006
+++ rc-scripts-0.4.0.24/rc.d/rc.sysinit	Sun Jan 15 08:43:26 2006
@@ -811,12 +811,12 @@
 if ! is_no "$NEED_XFILES" ; then
 	:>/var/run/utmpx
 	touch /var/log/wtmpx
-	chown root.utmp /var/run/utmpx /var/log/wtmpx
+	chown root:utmp /var/run/utmpx /var/log/wtmpx
 	chmod 0664 /var/run/utmpx /var/log/wtmpx
 else
 	:>/var/run/utmp
 	touch /var/log/wtmp
-	chown root.utmp /var/run/utmp /var/log/wtmp
+	chown root:utmp /var/run/utmp /var/log/wtmp
 	chmod 0664 /var/run/utmp /var/log/wtmp
 fi
 
