diff -urN rc-scripts-0.3.1.org/rc.d/rc.sysinit rc-scripts-0.3.1/rc.d/rc.sysinit
--- rc-scripts-0.3.1.org/rc.d/rc.sysinit	2004-01-17 00:13:42.164355620 +0100
+++ rc-scripts-0.3.1/rc.d/rc.sysinit	2004-01-17 00:15:53.224091932 +0100
@@ -375,11 +375,15 @@
 		show "Starting up RAID device %s" $RAIDDEV
 		busy
 		if [ -z "$RAIDSTAT" ]; then
-			# Try raidstart first...if that fails then
+			# Try mdadm, raidstart first...if that fails then
 			# fall back to raid0run and if that fails too
 			# fall back to raidadd, raidrun.
 			RESULT=1
-			if [ -x /sbin/raidstart ]; then
+			if [ -x /sbin/mdadm ]; then
+			    	/sbin/mdadm --assemble --scan
+			    	RESULT=$?
+			fi
+			if [ $RESULT -gt 0 -a -x /sbin/raidstart ]; then
 				/sbin/raidstart $i
 				RESULT=$?
 			fi
