Index: functions
===================================================================
--- rc-scripts/rc.d/init.d/functions	(revision 3049)
+++ rc-scripts/rc.d/init.d/functions	(working copy)
@@ -399,8 +399,8 @@
 #	elif [ -n "`$SH -c 'echo ${.sh.version}' 2>/dev/null`" ]; then
 	fi
 	while echo "$limits" | grep -q '[^ ]'; do
-		ulimit `echo ${limits} | awk '/^ *-[^-]*-.*$/ {sub(/^[^-]*-/,"");sub(/-.*$/,"");sub(/^.*$/,"-&");}{print;}'
-		limits="`echo ${limits} | awk '/^ *-[^-]*-.*$/ {sub(/^[^-]*-[^-]*/,"");}{print;}'
+		ulimit `echo ${limits} | sed -e 's/^ *\(-[^-]*\)\(-.*\)\?$/\1/'`
+		limits="`echo ${limits} | sed -e 's/^ *\(-[^-]*\)\(-.*\)\?$/\2/'`"
 	done
 
 	# And start it up.
