Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(wersja 12481)
+++ rc.d/rc.sysinit	(wersja 12485)
@@ -570,6 +570,10 @@
 	# Load modules
 	if ! use_upstart; then
 		load_kernel_modules modules
+		for f in /etc/modules-load.d/*.conf ; do
+			[ -r $f ] || continue
+			load_kernel_modules ${f##/etc/}
+		done
 	fi
 
 	if [ -x /sbin/multipath ] && ! is_no "$DM_MULTIPATH"; then
Index: service
===================================================================
--- service	(wersja 12481)
+++ service	(wersja 12485)
@@ -85,6 +85,14 @@
 		echo "${VERSION}" >&2
 		exit 0
 		;;
+	--ignore-dependencies)
+		export SYSTEMCTL_IGNORE_DEPENDENCIES=1
+		shift
+		;;
+	--skip-redirect)
+		export SYSTEMCTL_SKIP_REDIRECT=1
+		shift
+		;;
 	--upstart)
 		USE_UPSTART=yes
 		shift
@@ -116,7 +124,10 @@
 	esac
 done
 
-if [ -x "${SERVICEDIR}/${SERVICE}" ]; then
+if [ -x /bin/systemd_booted ] && /bin/systemd_booted && /bin/systemctl status ${SERVICE}.service > /dev/null 2>&1; then
+	echo >&2 "Redirecting to /bin/systemctl ${OPTIONS} ${SERVICE}.service"
+	exec /bin/systemctl ${ACTION} ${SERVICE}.service ${OPTIONS}
+elif [ -x "${SERVICEDIR}/${SERVICE}" ]; then
 	exec env -i USE_UPSTART=$USE_UPSTART LANG=$LANG PATH=$PATH TERM=$TERM "${SERVICEDIR}/${SERVICE}" ${ACTION} ${OPTIONS}
 else
 	echo "${SERVICE}: unrecognized service" >&2
Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(wersja 12506)
+++ rc.d/rc.sysinit	(kopia robocza)
@@ -173,7 +173,7 @@
 	fi
 
 	# we need /proc mounted before everything
-	is_fsmounted proc /proc || mount -n -o gid=17 -t proc /proc /proc
+	is_fsmounted proc /proc || mount -n /proc || mount -n -o gid=17,hidepid=2 -t proc /proc /proc
 
 	# Only read this once.
 	cmdline=$(cat /proc/cmdline)
