--- sysconfig/interfaces/data/chat-ppp0.tpsa.org	Mon Feb  5 18:57:58 2001
+++ sysconfig/interfaces/data/chat-ppp0.tpsa	Sun Apr 21 12:22:06 2002
@@ -1,5 +1,5 @@
 # (c) 1999 Grzegorz Stanislawski <stangrze@open.net.pl>
-#  $Id$
+#  $Id$
 # Script for dialing in to Public Internet Access by TPSA (0202122)
   
 ABORT		BUSY
@@ -10,7 +10,6 @@
 OK		$INITSTRING
 OK		atdt$PHONE
 CONNECT		\\c
+ppp\\r		\\c
 ame:		$USER
 ord:		$PASSWORD	
-ppp:		''
-starting:	''
