diff -urN rc-scripts-0.3.1.org/rc.d/rc.sysinit rc-scripts-0.3.1/rc.d/rc.sysinit
--- rc-scripts-0.3.1.org/rc.d/rc.sysinit	2003-10-16 22:52:57.000000000 +0200
+++ rc-scripts-0.3.1/rc.d/rc.sysinit	2003-10-16 22:59:09.000000000 +0200
@@ -356,8 +356,10 @@
 fi
 
 # Find and activate volume groups
-if [ -e /etc/lvmtab -a -x /sbin/vgscan -a -x /sbin/vgchange ]; then
+if [ -x /sbin/vgscan -a -x /sbin/vgchange ]; then
 	modprobe -s lvm-mod >/dev/null 2>&1
+	# device mapper (2.5+ and patched 2.4)
+	modprobe -s dm-mod >/dev/null 2>&1
 	run_cmd "Scanning for LVM volume groups" /sbin/vgscan && \
 	run_cmd "Activating LVM volume groups" /sbin/vgchange -a y
 fi
