diff -urN rc-scripts.orig/rc.d/rc.shutdown rc-scripts-0.4.0.2/rc.d/rc.shutdown
--- rc-scripts.orig/rc.d/rc.shutdown	2003-10-02 20:46:49.000000000 +0200
+++ rc-scripts-0.4.0.2/rc.d/rc.shutdown	2003-10-08 16:49:56.178983872 +0200
@@ -40,7 +40,7 @@
 sig=-15
 retry=3
 force=
-remaining=$(awk '!/(^#| proc | loopfs | devfs | devpts | shm | iso9660 | ramfs | tmpfs |^none|^\/dev\/root| \/ )/ {print $2}' /proc/mounts)
+remaining=$(awk '!/(^#| proc | loopfs | devfs | devpts | shm | iso9660 | ramfs | tmpfs | sysfs |^none|^\/dev\/root| \/ )/ {print $2}' /proc/mounts)
 while [ -n "$remaining" -a "$retry" -gt 0 ]
 do	
 	show "Unmounting file systems" ; busy
@@ -52,7 +52,7 @@
 	fi
 	
 	sleep 2
-	remaining=$(awk '!/(^#| proc | loopfs | devfs | devpts | shm |^none|^\/dev\/root| \/ )/ {print $2}' /proc/mounts)
+	remaining=$(awk '!/(^#| proc | loopfs | devfs | devpts | shm | sysfs |^none|^\/dev\/root| \/ )/ {print $2}' /proc/mounts)
 	[ -z "$remaining" ] && break
 	/sbin/fuser -k -m $sig $remaining > /dev/null
 	sleep 5
