--- rc-scripts-0.4.0.2/sysconfig/network-scripts/functions.network.orig	2003-10-02 20:46:49.000000000 +0200
+++ rc-scripts-0.4.0.2/sysconfig/network-scripts/functions.network	2003-10-12 15:36:26.392857840 +0200
@@ -64,7 +64,7 @@
 fi
 }
 
-# PLD GNU/Linux network functions
+# PLD Linux Distribution network functions
 # Copyright (C) 1999, 2000 Arkadiusz Mi¶kiewicz <misiek@pld-linux.org>
 
 # Setup Network Address Translation (NAT)
