--- lib/functions	2011-12-14 13:23:27.000000000 +0200
+++ lib/functions	2011-12-14 19:16:08.333233967 +0200
@@ -643,7 +643,7 @@
 		nice=${nice:-0}
 
 		# make nice level absolute, not to be dependant of nice level of shell where service started
-		nice=$((nice-$(nice)))
+		nice=$(($nice - $(nice)))
 
 		if [ "$closefds" = 1 ]; then
 			exec 1>&-
Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(wersja 12454)
+++ rc.d/rc.sysinit	(kopia robocza)
@@ -617,23 +617,26 @@
 		fi
 		if [ "$lvmversion" = "1" ] ; then
 			modprobe -s lvm-mod >/dev/null 2>&1
+			lvmignorelocking=""
 			lvmsysinit=""
 		elif [ "$lvmversion" = "2" ] ; then
 			modprobe -s dm-mod >/dev/null 2>&1
+			lvmignorelocking="--ignorelockingfailure"
 			lvmsysinit="--sysinit"
 		else
 			modprobe -s lvm-mod >/dev/null 2>&1
 			# device mapper (2.5+ and patched 2.4)
 			modprobe -s dm-mod >/dev/null 2>&1
+			lvmignorelocking=""
 			lvmsysinit=""
 		fi
 
-		run_cmd "Scanning for LVM volume groups" /sbin/vgscan $lvmsysinit
+		run_cmd "Scanning for LVM volume groups" /sbin/vgscan $lvmignorelocking
 		run_cmd "Activating LVM volume groups" /sbin/vgchange -a y $lvmsysinit
 		if [ "$lvmversion" = "2" ]; then
-			/sbin/vgmknodes $lvmsysinit
+			/sbin/vgmknodes $lvmignorelocking
 			# display VG statistics
-			/sbin/vgdisplay -s $lvmsysinit
+			/sbin/vgdisplay -s $lvmignorelocking
 		fi
 	fi
 
@@ -732,7 +735,7 @@
 			# LVM on RAID (keep in sync with LVM setting few lines above)
 			if [ "$golvm" -eq "1" ]; then
 				if [ -x /sbin/vgscan -a -x /sbin/vgchange ]; then
-					run_cmd "Scanning for LVM volume groups (on RAID)" /sbin/vgscan $lvmsysinit
+					run_cmd "Scanning for LVM volume groups (on RAID)" /sbin/vgscan $lvmignorelocking
 					run_cmd "Activating LVM volume groups (on RAID)" /sbin/vgchange -a y $lvmsysinit
 					[ "$lvmversion" = "2" ] && /sbin/vgmknodes
 				fi
