diff -ur rc-scripts-0.3.0.orig/src/ppp-watch.c rc-scripts-0.3.0/src/ppp-watch.c
--- rc-scripts-0.3.0.orig/src/ppp-watch.c	Tue May 15 18:03:41 2001
+++ rc-scripts-0.3.0/src/ppp-watch.c	Sun Dec 16 19:29:51 2001
@@ -78,7 +78,7 @@
 #include <glib.h>
 #include "shvar.h"
 
-#define IFCFGPREFIX "/etc/sysconfig/network-scripts/ifcfg-"
+#define IFCFGPREFIX "/etc/sysconfig/interfaces/ifcfg-"
 #define IFUP_PPP "/etc/sysconfig/network-scripts/ifup-ppp"
 
 static int theSigterm = 0;
Only in rc-scripts-0.3.0/src: ppp-watch.c~
diff -ur rc-scripts-0.3.0.orig/src/usernetctl.c rc-scripts-0.3.0/src/usernetctl.c
--- rc-scripts-0.3.0.orig/src/usernetctl.c	Tue May 15 18:03:41 2001
+++ rc-scripts-0.3.0/src/usernetctl.c	Sun Dec 16 19:32:24 2001
@@ -126,17 +126,17 @@
     if (argc != 3) usage();
 
     if (!strcmp(argv[2], "up")) {
-	cmd = "./ifup";
+	cmd = "/sbin/ifup";
     } else if (!strcmp(argv[2], "down")) {
-	cmd = "./ifdown";
+	cmd = "/sbin/ifdown";
     } else if (!strcmp(argv[2], "report")) {
 	report = 1;
     } else {
 	usage();
     }
 
-    if (chdir("/etc/sysconfig/network-scripts")) {
-	fprintf(stderr, "error switching to /etc/sysconfig/network-scripts: "
+    if (chdir("/etc/sysconfig/interfaces")) {
+	fprintf(stderr, "error switching to /etc/sysconfig/interfaces: "
 		"%s\n", strerror(errno));
 	exit(1);
     }
Only in rc-scripts-0.3.0/src: usernetctl.c~
