diff -urN aa/rc-scripts-0.4.0.15/rc.d/rc.sysinit rc-scripts-0.4.0.15/rc.d/rc.sysinit
--- aa/rc-scripts-0.4.0.15/rc.d/rc.sysinit	2004-08-20 14:16:03.000000000 +0200
+++ rc-scripts-0.4.0.15/rc.d/rc.sysinit	2004-09-18 12:03:38.056855848 +0200
@@ -27,6 +27,14 @@
     mount -n -o gid=17 -t sysfs sysfs /sys
 fi
 
+# udev support
+if [ -r /etc/sysconfig/udev ]; then
+  . /etc/sysconfig/udev
+  if is_yes "$USE_UDEV" && [ -x /sbin/start_udev ]; then
+    /sbin/start_udev
+  fi
+fi
+
 # selinux
 if grep -q selinuxfs /proc/filesystems && ! grep -q selinuxfs /proc/mounts; then
     mount -n -o gid=17 -t selinuxfs selinuxfs /selinux
