diff -urN rc-scripts-0.4.0.21.orig/lang.sh rc-scripts-0.4.0.21/lang.sh
--- rc-scripts-0.4.0.21.orig/lang.sh	2003-10-28 02:10:55.000000000 +0100
+++ rc-scripts-0.4.0.21/lang.sh	2005-11-10 22:51:23.231869250 +0100
@@ -7,11 +7,17 @@
     . /etc/sysconfig/i18n
 
    [ -n "$LANG" ] && export LANG || unset LANG
-   [ -n "$LC_CTYPE" ] && export LC_CTYPE || unset LC_CTYPE
+   [ -n "$LC_ADDRESS" ] && export LC_ADDRESS || unset LC_ADDRESS
    [ -n "$LC_COLLATE" ] && export LC_COLLATE || unset LC_COLLATE
+   [ -n "$LC_CTYPE" ] && export LC_CTYPE || unset LC_CTYPE
+   [ -n "$LC_IDENTIFICATION" ] && export LC_IDENTIFICATION || unset LC_IDENTIFICATION
+   [ -n "$LC_MEASUREMENT" ] && export LC_MEASUREMENT || unset LC_MEASUREMENT
    [ -n "$LC_MESSAGES" ] && export LC_MESSAGES || unset LC_MESSAGES
-   [ -n "$LC_NUMERIC" ] && export LC_NUMERIC || unset LC_NUMERIC
    [ -n "$LC_MONETARY" ] && export LC_MONETARY || unset LC_MONETARY
+   [ -n "$LC_NAME" ] && export LC_NAME || unset LC_NAME
+   [ -n "$LC_NUMERIC" ] && export LC_NUMERIC || unset LC_NUMERIC
+   [ -n "$LC_PAPER" ] && export LC_PAPER || unset LC_PAPER
+   [ -n "$LC_TELEPHONE" ] && export LC_TELEPHONE || unset LC_TELEPHONE
    [ -n "$LC_TIME" ] && export LC_TIME || unset LC_TIME
    [ -n "$LC_ALL" ] && export LC_ALL || unset LC_ALL
    [ -n "$LANGUAGE" ] && export LANGUAGE || unset LANGUAGE
