Index: rc.d/rc.sysinit
===================================================================
--- rc.d/rc.sysinit	(wersja 12018)
+++ rc.d/rc.sysinit	(kopia robocza)
@@ -272,7 +272,7 @@
 
 	# cgroup/cpuset support
 	if grep -q cgroup /proc/filesystems 2>/dev/null ; then
-		if ! grep -q cgroup /etc/fstab 2>/dev/null ; then
+		if ! grep -q "^[^#].*cgroup" /etc/fstab 2>/dev/null ; then
 			# creating is more convenient than artificial conflict with older udev
 			mkdir -p /dev/cgroup
 			# mount w/o options enables all available cgroup subsystems
@@ -280,7 +280,7 @@
 			mount -n -t cgroup none /dev/cgroup
 		fi
 	elif grep -q cpuset /proc/filesystems 2>/dev/null ; then
-		if ! grep -q cpuset /etc/fstab 2>/dev/null ; then
+		if ! grep -q "^[^#].*cpuset" /etc/fstab 2>/dev/null ; then
 			mount -n -t cpuset none /dev/cpuset
 		fi
 	fi
Index: rc.d/init.d/netfs
===================================================================
--- rc.d/init.d/netfs	(wersja 12019)
+++ rc.d/init.d/netfs	(kopia robocza)
@@ -15,6 +15,10 @@
 . /etc/rc.d/init.d/cryptsetup
 . /etc/sysconfig/network
 
+if is_yes "$VSERVER"; then
+	exit 0
+fi
+
 # nfs uses own script for that
 if [ ! -f /etc/rc.d/init.d/nfsfs ]; then
 	NFSFSTAB=$(LC_ALL=C awk '!/^#/ && $3 ~ /^nfs/ && $3 != "nfsd" && $4 !~ /noauto/ { print $2 }' /etc/fstab)
