--- ./rc.d/init.d/network.org	Fri Nov  9 19:34:41 2001
+++ ./rc.d/init.d/network	Tue Feb 12 15:31:45 2002
@@ -76,7 +76,7 @@
 		else
 			IPXAUTOPRIMARY="off"
 		fi
-		/usr/bin/ipx_configure --auto_primary=$IPXAUTOPRIMARY
+		/sbin/ipx_configure --auto_primary=$IPXAUTOPRIMARY
 	fi
 	if [ -n $IPXAUTOFRAME ] ; then
 		if is_yes "$IPXAUTOFRAME"; then
@@ -84,10 +84,10 @@
 		else
 			IPXAUTOFRAME="off"
 		fi
-		/usr/bin/ipx_configure --auto_interface=$IPXAUTOFRAME
+		/sbin/ipx_configure --auto_interface=$IPXAUTOFRAME
 	fi
 	if [ -n "$IPXINTERNALNETNUM" -a "$IPXINTERNALNETNUM" != "0" ]; then
-		/usr/bin/ipx_internal_net add $IPXINTERNALNETNUM $IPXINTERNALNODENUM
+		/sbin/ipx_internal_net add $IPXINTERNALNETNUM $IPXINTERNALNODENUM
 	fi
 fi
 }
@@ -204,6 +204,11 @@
 	;;
   restart)
 	$0 stop
+	show "Waiting 5 seconds for interfaces shutdown"
+	busy
+	sleep 5
+	deltext
+	ok
 	$0 start
 	;;
   *)
--- ./sysconfig/network-scripts/ifup-ipx.org	Wed May 16 15:12:07 2001
+++ ./sysconfig/network-scripts/ifup-ipx	Tue Feb 12 15:29:18 2002
@@ -19,8 +19,8 @@
 	exit 0
 fi
 
-if [ ! -x /usr/bin/ipx_interface ] ; then
-	nls "%s is missing. Can't continue." "/usr/bin/ipx_interface"
+if [ ! -x /sbin/ipx_interface ] ; then
+	nls "%s is missing. Can't continue." "/sbin/ipx_interface"
 	exit 1
 fi
 
@@ -44,7 +44,7 @@
 	       esac
 
 	       ip link set $DEVICE up
-	       /usr/bin/ipx_interface add $primary $DEVICE $frametype \
+	       /sbin/ipx_interface add $primary $DEVICE $frametype \
 	           $(eval echo $(echo \$`echo IPXNETNUM_$framename`))
 	       ;;
 	esac
