diff -urN rc-scripts-0.4.0.10.org/rc.d/rc.sysinit rc-scripts-0.4.0.10/rc.d/rc.sysinit
--- rc-scripts-0.4.0.10.org/rc.d/rc.sysinit	2004-05-17 02:20:31.352691184 +0200
+++ rc-scripts-0.4.0.10/rc.d/rc.sysinit	2004-05-17 02:20:56.136923408 +0200
@@ -451,7 +451,7 @@
 fi
 # LVM
 if is_yes "$EVMS_LVM" || [ -x /sbin/vgscan -a -x /sbin/vgchange ]; then
-	lvmversion=$(/sbin/vgchange --version 2>/dev/null|head -n 1|awk '{gsub("vgchange: Logical Volume Manager ",""); gsub("LVM version:     ",""); gsub(/\..*/,"");print $1}')
+	lvmversion=$(/sbin/vgchange --version 2>/dev/null|awk '{gsub("vgchange: Logical Volume Manager ",""); gsub("LVM version:     ",""); gsub(/\..*/,"");print $1; exit}')
    	if [ "$lvmversion" = "1" ] ; then
 		modprobe -s -k lvm-mod >/dev/null 2>&1
 		lvmignorelock=""
