diff -urN rc-scripts-0.3.1.org/sysconfig/Makefile.am rc-scripts-0.3.1/sysconfig/Makefile.am
--- rc-scripts-0.3.1.org/sysconfig/Makefile.am	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/Makefile.am	Sun Aug 11 16:03:34 2002
@@ -10,6 +10,7 @@
 	network \
 	static-routes \
 	static-nat \
+	static-arp \
 	i18n \
 	timezone
 
diff -urN rc-scripts-0.3.1.org/sysconfig/Makefile.in rc-scripts-0.3.1/sysconfig/Makefile.in
--- rc-scripts-0.3.1.org/sysconfig/Makefile.in	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/Makefile.in	Sun Aug 11 16:03:44 2002
@@ -93,6 +93,7 @@
 	network \
 	static-routes \
 	static-nat \
+	static-arp \
 	i18n \
 	timezone
 
diff -urN rc-scripts-0.3.1.org/sysconfig/network-scripts/Makefile.am rc-scripts-0.3.1/sysconfig/network-scripts/Makefile.am
--- rc-scripts-0.3.1.org/sysconfig/network-scripts/Makefile.am	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/network-scripts/Makefile.am	Sun Aug 11 16:10:32 2002
@@ -18,6 +18,7 @@
 	ifup-aliases \
 	ifup-br \
 	ifup-ipx \
+	ifup-neigh \
 	ifup-plip \
 	ifup-post \
 	ifup-ppp \
diff -urN rc-scripts-0.3.1.org/sysconfig/network-scripts/Makefile.in rc-scripts-0.3.1/sysconfig/network-scripts/Makefile.in
--- rc-scripts-0.3.1.org/sysconfig/network-scripts/Makefile.in	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/network-scripts/Makefile.in	Sun Aug 11 16:10:44 2002
@@ -101,6 +101,7 @@
 	ifup-aliases \
 	ifup-br \
 	ifup-ipx \
+	ifup-neigh \
 	ifup-plip \
 	ifup-post \
 	ifup-ppp \
diff -urN rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-neigh rc-scripts-0.3.1/sysconfig/network-scripts/ifup-neigh
--- rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-neigh	Thu Jan  1 01:00:00 1970
+++ rc-scripts-0.3.1/sysconfig/network-scripts/ifup-neigh	Sun Aug 11 16:13:52 2002
@@ -0,0 +1,46 @@
+#!/bin/sh
+#
+#	$Id$
+#
+# adds static arps for device $DEVICE
+
+. /etc/sysconfig/network
+. /etc/rc.d/init.d/functions
+. /etc/sysconfig/network-scripts/.functions
+
+DEV=$1
+
+if [ -z "$DEV" ]; then
+	nls "Usage: %s <device name>" "ifup-neigh" >&2
+	exit 1
+fi
+
+if [ ! -f /etc/sysconfig/static-arp ]; then
+	exit 0
+fi
+
+if [ -f "/etc/sysconfig/interfaces/ifcfg-$DEV" ] ; then
+        CONFIG="/etc/sysconfig/interfaces/ifcfg-$DEV"
+else
+        CONFIG="$DEV"
+fi
+		
+source_config
+
+# set all major variables
+setup_ip_param
+
+get_ppp_device_and_pid
+
+# note the trailing white space character in the grep gets rid of aliases
+egrep "^($DEVICE|any)[[:blank:]]" /etc/sysconfig/static-arp | while read iface mac ip state args; do
+	if is_no "$IPV4_NETWORKING"; then
+		continue
+	fi
+	[ -z "$state" ] && state="stale"
+	/sbin/ip neigh replace $ip lladdr $mac nud $state dev $DEVICE
+done
+
+
+# This must be last line !
+# vi:syntax=sh:tw=78:ts=8:sw=4
diff -urN rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-post rc-scripts-0.3.1/sysconfig/network-scripts/ifup-post
--- rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-post	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/network-scripts/ifup-post	Sun Aug 11 16:05:01 2002
@@ -27,6 +27,7 @@
 
 /etc/sysconfig/network-scripts/ifup-aliases ${CONFIG}
 /etc/sysconfig/network-scripts/ifup-routes ${CONFIG}
+/etc/sysconfig/network-scripts/ifup-neigh ${CONFIG}
 
 # don't set hostname on ppp/slip connections
 if [ -n "$NEEDHOSTNAME" -a "${DEVICETYPE}" != "ppp" -a "${DEVICETYPE}" != "slip" ]; then
diff -urN rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-routes rc-scripts-0.3.1/sysconfig/network-scripts/ifup-routes
--- rc-scripts-0.3.1.org/sysconfig/network-scripts/ifup-routes	Sun Aug 11 16:03:09 2002
+++ rc-scripts-0.3.1/sysconfig/network-scripts/ifup-routes	Sun Aug 11 16:04:28 2002
@@ -33,7 +33,7 @@
 get_ppp_device_and_pid
 
 # note the trailing white space character in the grep gets rid of aliases
-grep "^$DEVICE[[:blank:]]" /etc/sysconfig/static-routes | while read device args; do
+egrep "^($DEVICE|any)[[:blank:]]" /etc/sysconfig/static-routes | while read device args; do
 	if (echo $args | grep -q ":") then
 		if is_no "$IPV6_NETWORKING"; then
 			continue
diff -urN rc-scripts-0.3.1.org/sysconfig/static-arp rc-scripts-0.3.1/sysconfig/static-arp
--- rc-scripts-0.3.1.org/sysconfig/static-arp	Thu Jan  1 01:00:00 1970
+++ rc-scripts-0.3.1/sysconfig/static-arp	Sun Aug 11 16:00:09 2002
@@ -0,0 +1,3 @@
+# dev mac ip state opts
+# eth0 00:80:48:12:c2:3c 192.168.10.10 permanent
+#eth1 00:80:48:12:c2:44 12.12.12.12 permanent
