--- rc-scripts-0.4.2.6/rc.d/rc.sysinit	2009-12-02 21:16:05.894346123 +0200
+++ rc-scripts-0.4.2.6/rc.d/rc.sysinit	2009-12-02 21:13:38.664359933 +0200
@@ -261,13 +261,13 @@
 	fi
 
 	CLOCKDEF=""
-	CLOCKFLAGS="-w"
+	CLOCKFLAGS="--hctosys"
 
 	if is_yes "$UTC" ; then
-		CLOCKFLAGS="$CLOCKFLAGS -u"
+		CLOCKFLAGS="$CLOCKFLAGS --utc"
 		CLOCKDEF="$CLOCKDEF (utc)"
 	else
-		CLOCKFLAGS="$CLOCKFLAGS -l"
+		CLOCKFLAGS="$CLOCKFLAGS --localtime"
 		CLOCKDEF="$CLOCKDEF (local)"
 	fi
 
